test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJps.java

```diff
@@ -0,0 +1,169 @@
+		
+		searchResult = search(tgtMgr1.targetId, jpsOutput);
+		assertTrue(CHILD_IS_MISSING, search(tgtMgr1.targetId, jpsOutput).isPresent());
+		assertEquals(SPURIOUS_OUTPUT, tgtMgr1.targetId, searchResult.get());
+		
+		/* now try with non-default vm ID */
+		searchResult = search(tgtMgr2.targetId, jpsOutput);
+		assertTrue(CHILD_IS_MISSING, search(tgtMgr2.targetId, jpsOutput).isPresent());
+		assertEquals(SPURIOUS_OUTPUT, tgtMgr2.targetId, searchResult.get());
+		
+		tgtMgr1.terminateTarget();
+		tgtMgr2.terminateTarget();
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testApplicationArguments() throws IOException {
+		if (skipTest) {
+			return;
+		}
+		List<String> targetArgs = Arrays.asList("foo", "bar");  //$NON-NLS-1$//$NON-NLS-2$
+		TargetManager tgtMgr = new TargetManager(TARGET_VM_CLASS, null, targetArgs);
+		assertTrue(CHILD_PROCESS_DID_NOT_LAUNCH, tgtMgr.syncWithTarget());
+		List<String> jpsOutput = runCommand(Collections.singletonList("-m")); //$NON-NLS-1$
+		Optional<String> searchResult =  search(tgtMgr.targetId, jpsOutput);
+		assertTrue(CHILD_IS_MISSING, search(tgtMgr.targetId, jpsOutput).isPresent());
+		for (String a: targetArgs) {
+			assertTrue(MISSING_ARGUMENT + a, searchResult.get().contains(a));
+		}
+		tgtMgr.terminateTarget();
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testJvmArguments() throws IOException {
+		if (skipTest) {
+			return;
+		}
+		List<String> vmArgs = Collections.singletonList("-Dfoo=bar"); //$NON-NLS-1$
+		TargetManager tgtMgr = new TargetManager(TARGET_VM_CLASS, null, null, vmArgs, null);
+		assertTrue(CHILD_PROCESS_DID_NOT_LAUNCH, tgtMgr.syncWithTarget());
+		List<String> jpsOutput = runCommand(Collections.singletonList("-v")); //$NON-NLS-1$
+		Optional<String> searchResult =  search(tgtMgr.targetId, jpsOutput);
+		assertTrue(CHILD_IS_MISSING, search(tgtMgr.targetId, jpsOutput).isPresent());
+		for (String a: vmArgs) {
+			assertTrue(MISSING_ARGUMENT + a, searchResult.get().contains(a));
+		}
+		tgtMgr.terminateTarget();
+	}
+
+	/**
+	 * Look for a string in a list of strings using case sensitive substring searching
+	 * @param needle string for which to search
+	 * @param haystack list of strings in which to search
+	 * @return first string in haystack containing needle
+	 */
+	static Optional<String> search(String needle, List<String> haystack) {
+		return haystack.stream().filter(s -> s.contains(needle)).findFirst();
+	}
+
+	@BeforeMethod
+	protected void setUp(Method testMethod) {
+		testName = testMethod.getName();
+		log("starting " + testName);		 //$NON-NLS-1$
+	}
+	
+	/**
+	 * Don't run the test if it is running on non-OpenJ9 (e.g. IBM) Java
+	 */
+	@BeforeSuite
+	void setupSuite() {
+		commandName = JPS_COMMAND;
+		assertTrue("Attach API failed to launch", TargetManager.waitForAttachApiInitialization()); //$NON-NLS-1$
+		vmId = TargetManager.getVmId();
+		skipTest = !System.getProperty("java.vm.vendor").contains("OpenJ9");  //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
Is this really required? I'm not sure that it is, since IBM isn't listed as an `<impl>` in the playlist.